cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -0,0 +1,528 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :description: HTTP RESTful Interface to the Cask Data Application Platform
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _http-restful-api-metadata:
+.. _http-restful-api-v3-metadata:
+
+=========================
+Metadata HTTP RESTful API
+=========================
+
+.. highlight:: console
+
+.. include:: ../../../_common/_includes/include-v320-beta-metadata.rst
+
+Use the CDAP Metadata HTTP RESTful API to set, retrieve, and delete the metadata annotations
+of applications, datasets, streams, and other elements in CDAP.
+
+Metadata consists of **properties** (a list of key-value pairs) or **tags** (a list of keys).
+Metadata and their use are described in the :ref:`Developers' Manual: Metadata <metadata>`.
+
+The HTTP RESTful API is divided into three sections:
+
+- :ref:`metadata properties <http-restful-api-metadata-properties>`;
+- :ref:`metadata tags <http-restful-api-metadata-tags>`; and
+- :ref:`searching properties and viewing lineage <http-restful-api-metadata-searching-viewing>`
```

Comment:
I think it would be good to split search and lineage here. They are different entities.